src/types.rs

```diff
@@ -26,4 +31,133 @@ pub enum Authority {
   ManagedNode,    // in our group and routing table
   ManagedClient,  // in our group
   Client,         // detached
+  Unknown
+}
+
+impl Encodable for Authority {
+  fn encode<E: Encoder>(&self, e: &mut E)->Result<(), E::Error> {
+    let mut authority = "";
+    match *self {
+      Authority::ClientManager => authority = "ClientManager",
+      Authority::NaeManager => authority = "NaeManager",
+      Authority::NodeManager => authority = "NodeManager",
+      Authority::ManagedNode => authority = "ManagedNode",
+      Authority::ManagedClient => authority = "ManagedClient",
+      Authority::Client => authority = "Client",
+      Authority::Unknown => authority = "Unknown",
+    }
+    CborTagEncode { tag : 5483_000 , data : &authority }.encode(e)
```

Comment:
I think this tag needs to be unique across all libs. We need to sort out a scheme for our own libs, but this number is used in types. Perhaps routing should use 5483_1XXX or similar